src/lib/TopLevel.hs

```diff
@@ -593,10 +590,11 @@ execUDecl mname decl = do
 
 compileTopLevelFun
   :: (Topper m, Mut n)
-  => AtomName n -> NaryPiType n -> Atom n -> [Atom n] -> m n ()
-compileTopLevelFun fname naryPi f staticArgs = do
-  f' <- forceDeferredInlining f
-  fSimp <- simplifyTopFunction naryPi f' staticArgs
+  => AtomName n -> m n ()
+compileTopLevelFun fname = do
+  TopFunBound fTy (SpecializedTopFun s) <- lookupAtomName fname
+  fPreSimp <- specializedFunPreSimpDefinition fTy s
```

Comment:
It feels weird to do the look up above and then pass in `fTy` and `s` below (what if you pass in a type that doesn't match the specialization?!). Perhaps move the lookup into `specializedFunPreSimpDefinition`?